tooltip label should now be optional

mrjoh3 opened the pull request:
Hi Joshua, 
I hope this pull request is not too confusing. I am new to github and have a lot to learn. 

The purpose of this pull request is to change the way the tooltip label works. As it was everything would break if a character vector of labels was not included. 

I have also been working on a "mousedown" function that opens a url in the format word.html. This works well  for my usage but I feel is currently a bit limited so it is commented out. 

sorry for the confusion, 

Matt

mrjoh3 committed:
use provided color instead of scale

Hi Joshua, 
I was having trouble with the colors whenever I provided a color for each word if the word freq was the same for two or more words. It traced back to the behavior of d3.scale.ordinal() line 78. The change I have made is just to check for every_word_has_own_color and use the d.color if it is present. 

This solves my problem but does not necessarily behave in the way you want it to but please have a look. 

Matt

mrjoh3 committed:
made label variable optional

only define a label column in the "data" data.frame if  the label parameter is not null (see row 124).  Without this change label is a required field and this was not the intention.

Commit:
added option to open url for each word. currently very limited

Commit:
Merge branch 'patch-1' of https://github.com/mrjoh3/d3wordcloud

Commit:
Merge branch 'patch-2' of https://github.com/mrjoh3/d3wordcloud

Commit:
added label to x.pars.label and added x.pars.label to tooltip mouseover

Commit:
patch to complete tooltip label functionality

Commit:
commented out mousedown function

jbkunst commented:
Hi @mrjoh3 !

Thanks so much, I'm so sorry because I have enough time to this project yet. So I'll accept you pull request.

If you check https://travis-ci.org/jbkunst/d3wordcloud/builds/125967855 there some mismatches in the documentation. Can you fix this?

Finnaly can you make some minimal examples (in rpubs) to view the new funcionalities?

Thanks again. 

Aps! Almost forgot. If you make a new pullrequest, please add you as ctb in the DESCRIPTION file ;).

Commit:
updated documentation to include label

Commit:
added MJ as contributor

Commit:
added label to readme file

mrjoh3 commented:
Hi Joshua, 

I have updated the documentation and added myself as a contributor. The travisci now seems happy.

I am happy to add examples to the rpubs examples but I am afraid I could not work out which file to add the examples to. Can you give me a rough outline of how to do this and I should be able to get it dome by the weekend.

thanks

Matt
